assisted-events-scrape/events_scrape/events_scrape.py

```diff
@@ -38,250 +26,69 @@
+        self.changes = Changes()
+        self.error_counter = ErrorCounter()
 
-    def process_and_log_events(self, cluster_bash_data, event_list, event_names, only_new_events=True):
-        for event in event_list[::-1]:
-            if process.is_event_skippable(event):
-                continue
+    def start_workers(self, config: ScraperConfig) -> None:
 
-            doc_id = get_doc_id(event)
-            cluster_bash_data["no_name_message"] = get_no_name_message(event["message"], event_names)
-            cluster_bash_data["inventory_url"] = self.inventory_url
+        self.clusters_queue = Queue()
+        for n in range(config.n_workers):
+            workers_config = ClusterEventsWorkerConfig(
+                self.clusters_queue,
+                f"Worker-{n}",
+                config.sentry,
+                self.error_counter,
+                self.changes
+            )
 
-            if "props" in event:
-                event["event.props"] = json.loads(event["props"])
+            worker = ClusterEventsWorker(workers_config,
+                                         cluster_repository=self.cluster_repo,
+                                         event_repository=self.event_repo,
+                                         cluster_events_storage=self.cluster_events_storage)
 
-            process_event_doc(event, cluster_bash_data)
-            ret = self.log_doc(cluster_bash_data, doc_id)
+            worker.daemon = True
+            worker.start()
 
-            for key in event:
-                _ = cluster_bash_data.pop(key, None)
-
-            if not ret and only_new_events:
-                break
-
-    def save_new_backup(self, cluster_id, event_list, metadata_json):
-        cluster_backup_directory_path = os.path.join(self.backup_destination, f"cluster_{cluster_id}")
-        if not os.path.exists(cluster_backup_directory_path):
-            os.makedirs(cluster_backup_directory_path)
+    def is_idle(self):
+        return not self.changes.has_changed_in_last_minutes(self.max_idle_minutes)
 
-        event_dest = os.path.join(cluster_backup_directory_path, "events.json")
-        with open(event_dest, "w") as f:
-            json.dump(event_list, f, indent=4)
+    def has_too_many_unexpected_errors(self):
+        return self.error_counter.get_errors() > self.errors_before_restart
 
-        metadata_dest = os.path.join(cluster_backup_directory_path, "metadata.json")
-        with open(metadata_dest, "w") as f:
-            json.dump(metadata_json, f, indent=4)
+    def run_service(self):
+        clusters = self.cluster_repo.list_clusters()
 
-    def log_doc(self, doc, id_):
-        try:
-            res = self.es.create(index=self.index, body=doc, id=id_)
-        except elasticsearch.exceptions.ConflictError:
-            log.debug("Hit logged event")
+        if not clusters:
+            log.warning(f'No clusters were found, waiting {RETRY_INTERVAL / 60} min')
+            time.sleep(RETRY_INTERVAL)
```

Comment:
sleep here is a little bit strange for me as there is not loop and i needed to go to the calling function to understand why it was needed

Reply:
Totally agree, during the next iteration we'll try to organize better this part as well.

Reply:
Actually I'll just change it, it only takes 1 minute